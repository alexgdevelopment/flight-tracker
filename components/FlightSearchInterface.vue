<template>
  <div>
    <v-text-field
      v-model="searchValue"
      clearable
      label="Flight number"
      autofocus
      class="px-4"
      :counter="10"
      :rules="searchFieldRules"
      @click:clear="onClear"
    ></v-text-field>
    <v-menu
      v-model="datePicker"
      :close-on-content-click="true"
      transition="scroll-y-transition"
      :nudge-right="30"
      offset-y
      min-width="auto"
    >
      <template #activator="{ on, attrs }">
        <v-text-field
          v-show="showDatePicker"
          v-model="searchDate"
          prepend-icon="mdi-calendar"
          readonly
          v-bind="attrs"
          v-on="on"
        ></v-text-field>
      </template>
      <v-date-picker
        v-model="searchDate"
        :color="mainColor"
        show-adjacent-months
        @input="datePicker = false"
      >
      </v-date-picker>
    </v-menu>
    <v-row v-if="showSearchButton" class="mt-3" justify="center">
      <v-col cols="auto">
        <v-btn @click="buttonClick">Go</v-btn>
      </v-col>
    </v-row>
    <v-col v-if="noResultsInSearch">
      <p>No results.</p>
    </v-col>
    <v-col
      v-else-if="flights"
      class="results-container"
      data-testid="flightsContainer"
    >
      <FlightCard
        v-for="flight in flights"
        :key="flight.number"
        :flight="flight"
      />
    </v-col>
  </div>
</template>

<script>
  import { mapState } from 'vuex'
  export default {
    data() {
      return {
        datePicker: false,
        searchFieldRules: [
          (v) => v === null || v.length > 1 || 'Min 2 characters',
          (v) => v === null || v.length <= 10 || 'Max 10 characters',
        ],
      }
    },
    computed: {
      ...mapState({
        mainColor: (state) => state.mainColor,
        flights: (state) => state.flights.flights,
      }),
      noResultsInSearch() {
        return this.flights?.length === 0
      },
      searchValue: {
        get() {
          return this.$store.state.searchStringParam
        },
        set(value) {
          this.$store.commit('setSearchStringParam', { searchStringParam: value })
        },
      },
      searchDate: {
        get() {
          return this.$store.state.searchDate
        },
        set(value) {
          this.$store.commit('setSearchDate', { searchDate: value })
        },
      },

      showDatePicker() {
        return this.searchValue !== '' && this.searchValue !== null
      },

      showSearchButton() {
        return (
          this.searchValue !== '' && this.searchValue !== null && this.searchDate
        )
      },
    },

    methods: {
      buttonClick() {
        this.$store.dispatch('search')
      },
      onClear() {
        this.$store.dispatch('clearSearch')
      },
    },
  }
</script>