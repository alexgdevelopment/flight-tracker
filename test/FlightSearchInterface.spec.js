import { mount } from '@vue/test-utils'
import { Store } from 'vuex'
import { mockFlight } from './mockObjects'
import FlightSearchInterface from '@/components/FlightSearchInterface.vue'

describe('FlightSearchInterface', () => {
  let wrapper = null
  let store
  let state

  beforeEach(() => {
    state = {
      mainColor: 'green',
    }
    store = new Store({
      state,
      modules: {
        flights: {
          arrivals: [mockFlight],
          departures: [mockFlight],
        },
      },
    })
    wrapper = mount(FlightSearchInterface, {
      store,
      stubs: {
        'v-text-field': {
          template: '<div class="v-text-field" />',
        },
        FlightCard: {
          template: '<div class="flight-card" />',
        },
      },
    })
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
