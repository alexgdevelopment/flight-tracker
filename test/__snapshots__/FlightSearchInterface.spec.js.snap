// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightSearchInterface should match the snapshot 1`] = `
<div>
  <div
    autofocus="autofocus"
    class="v-text-field px-4"
    clearable=""
    counter="10"
    label="Flight Number"
    rules="(v) => {
          /* istanbul ignore next */
          cov_evj73t90m().f[2]++;
          cov_evj73t90m().s[6]++;
          return (
            /* istanbul ignore next */
            (cov_evj73t90m().b[0][0]++, v === null) ||
            /* istanbul ignore next */
            (cov_evj73t90m().b[0][1]++, v.length > 1) ||
            /* istanbul ignore next */
            (cov_evj73t90m().b[0][2]++, 'Min 2 characters')
          );
        },(v) => {
          /* istanbul ignore next */
          cov_evj73t90m().f[3]++;
          cov_evj73t90m().s[7]++;
          return (
            /* istanbul ignore next */
            (cov_evj73t90m().b[1][0]++, v === null) ||
            /* istanbul ignore next */
            (cov_evj73t90m().b[1][1]++, v.length <= 10) ||
            /* istanbul ignore next */
            (cov_evj73t90m().b[1][2]++, 'Max 10 characters')
          );
        }"
  />
   
  <div
    class="v-menu"
  >
    <!---->
  </div>
  <div
    aria-expanded="false"
    aria-haspopup="true"
    class="v-text-field"
    prepend-icon="mdi-calendar"
    readonly="readonly"
    role="button"
  />
   
  <div
    class="row mt-3 justify-center"
    style="display: none;"
  >
    <div
      class="col col-auto"
    >
      <button
        class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          Go
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;
